app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -43,6 +43,20 @@ def optional_authorize
       @repository = find_scope_repository
       if @repository && (@repository.environment.registry_unauthenticated_pull || ssl_client_authorized?(@repository.organization.label))
         true
+      elsif params['action'] == 'catalog'
+        token = request.headers['Authorization']
+        if token
+          token_type, token = token.split
+          if token_type == 'Bearer' && token
+            personal_token = PersonalAccessToken.find_by_token(token)
+            if personal_token && !personal_token.expired?
+              User.current = User.unscoped.find(personal_token.user_id)
+            end
+          end
```

Comment:
This looks heavily duplicated from the registry_authorize method, could we combine these two into one?  (I also notice this isn't handling the 'basic' token, whatever that is, so one method that handles all tokens (optionally) would be good i think

Reply:
It is duplicated and consolidating the logic would be a good thing.